release notes 23.10 (#906)
akka/akka.io: blog/_posts/2023-10-31-akka-23.10-released.md

---
layout: post
title: Akka 23.10 Released
author: Patrik Nordwall
short: We are excited to announce Akka 23.10 release
category: news
tags: [releases]
---

Dear hakkers,

Today we are excited to announce the availability of the Akka 23.10 release. It consists of the following modules:

* Akka (core) 2.9.0
* Akka HTTP 10.6.0
* Akka gRPC 2.4.0
* Akka Management 1.5.0
* Alpakka Kafka 5.0.0
* Alpakka 7.0.0
* Akka Persistence R2DBC 1.2.0
* Akka Persistence JDBC 5.3.0
* Akka Persistence Cassandra 1.2.0
* Akka Projections 1.5.0
* Akka Diagnostics 2.1.0
* Akka Insights 2.19.0 (to be released soon)

> Starting from this release, the Akka releases are now published to our own dedicated repository. This change provides us with more control over the distribution process, ensuring faster and more secure updates for our users. The repository is added to your build configuration as described in the [migration guide](https://doc.akka.io/docs/akka/current/project/migration-guide-2.8.x-2.9.x.html#migration-guide-2-8-x-to-2-9-x).

Reviewer: ```suggestion
Starting from this release, the Akka releases are now published to our own dedicated repository. This change provides us with more control over the distribution process, ensuring faster and more secure updates for our users. The repository needs to be added to your build configuration as described in the [migration guide](https://doc.akka.io/docs/akka/current/project/migration-guide-2.8.x-2.9.x.html#migration-guide-2-8-x-to-2-9-x).
```

## New features

For deeper insight into some of these new features, please visit the [Lightbend blog](https://www.lightbend.com/blog), which will publish articles about the features during the next month.

### Akka Edge

> [Akka Edge](https://doc.akka.io/docs/akka-edge/current/index.html) is a set of features that will help you with stretching an Akka system to the Edge. Akka Edge is similar to Akka Distributed Cluster and they share features and implementation, but Akka Edge is intended for use cases with even more geographically distributed services and possibly in more resource constrained environments.

Reviewer 2: ```suggestion
[Akka Edge](https://doc.akka.io/docs/akka-edge/current/index.html) is a set of features that will help you with stretching an Akka system to the Edge. Akka Edge is similar to [Akka Distributed Cluster](https://doc.akka.io/docs/akka-distributed-cluster/current/index.html) and they share features and implementation, but Akka Edge is intended for use cases with even more geographically distributed services and possibly in more resource constrained environments.
```

It comes with new features such as:

* Projections over gRPC for the Edge - asynchronous brokerless service-to-service communication.
* Connections are always established from the edge service.
* Scalability and efficiency improvements to handle the large scale of many edge services.
* Support for more constrained environments such as running with GraalVM native image and lightweight Kubernetes distributions.
* Support for multidimensional autoscaling and scale to near zero.

### H2 dialect in R2DBC persistence plugin

Support for the lightweight H2 database has been added to Akka Persistence R2DBC and Akka Projections. H2 can be useful as in-memory database for testing or as a lightweight database for single node systems.

### Starting projection from snapshot

> The [Projection can use snapshots](https://doc.akka.io/docs/akka-projection/current/eventsourced.html#sourceprovider-for-eventsbyslicesstartingfromsnapshots) as starting points and thereby reducing number of events that have to be loaded. This can be useful if the consumer starts from zero or if it has been disconnected for a long while.

Reviewer 2: ```suggestion
The [projection can use snapshots](https://doc.akka.io/docs/akka-projection/current/eventsourced.html#sourceprovider-for-eventsbyslicesstartingfromsnapshots) as starting points and thereby reducing number of events that have to be loaded. This can be useful if the consumer starts from zero or if it has been disconnected for a long while.
```

### Scalability improvements for many projections
[Many projection consumers](https://doc.akka.io/docs/akka-projection/current/eventsourced.html#many-projections) of the same events can share the same event stream from the database.

## Security Updates

We run automated security vulnerability and license policy scanning of dependencies. Several dependencies have been updated to fix security vulnerabilities and keep up with latest improvements.

The following CVE’s are addressed in this build.

* [CVE-2023-44487](https://akka.io/security/akka-http-cve-2023-44487.html)
* [CVE-2023-45865](https://akka.io/security/akka-cve-2023-45865.html)

> * [CVE-2022-3171](https://nvd.nist.gov/vuln/detail/CVE-2022-3171.html)

Reviewer 2: Looks like possibly none of the nvd.nist.gov links works, you end up on an error page

Reviewer: The `.html` should go `https://nvd.nist.gov/vuln/detail/CVE-2022-3171` works (but slow).

Author: my bad, should be without `.html`
